server/game.py

```diff
@@ -282,11 +274,25 @@ def send_cooldown(self, nickname, death_reason):
         """Remove a train and update game size"""
         if nickname in self.trains:
             # Register the death time
-            self.dead_trains[nickname] = time.time()
+            # In grading mode, use tick-based cooldown
+            self.train_death_ticks[nickname] = self.current_tick
+            
+            # Calculate the expected respawn tick based on the standard tickrate
+            standard_tickrate = 60.0  # Reference tickrate
+            tickrate_ratio = standard_tickrate / self.config.tick_rate
+            
+            # For tickrate < standard (e.g. 30), the ratio > 1, making cooldown longer in real time
+            # For tickrate > standard (e.g. 240), the ratio < 1, making cooldown shorter in real time
+            adjusted_cooldown_ticks = int(self.config.respawn_cooldown_seconds * self.config.tick_rate * tickrate_ratio)
+            expected_respawn_tick = self.current_tick + adjusted_cooldown_ticks
+            
+            real_seconds = adjusted_cooldown_ticks / self.config.tick_rate
+            logger.debug(f"Train {nickname} died at tick {self.current_tick}, reason: {death_reason}")
+            logger.debug(f"Expected respawn at tick {expected_respawn_tick} (after {adjusted_cooldown_ticks} ticks, {real_seconds:.2f}s real time)")
 
             # Clean up the last delivery time for this train
-            if nickname in self.last_delivery_times:
-                del self.last_delivery_times[nickname]
+            if nickname in self.last_delivery_ticks:
+                del self.last_delivery_ticks[nickname]
```

Comment:
nit: `self.last_delivery_ticks.pop(nickname, None)` deletes nickname and doesn't need the `if` check.